sdk/core/Azure.Core/src/Serialization/ModelJsonConverter.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// .
+    /// </summary>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+    public class ModelJsonConverter : JsonConverter<IJsonSerializable>
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+    {
+        /// <summary>
+        /// .
+        /// </summary>
+        public bool IgnoreAdditionalProperties { get; }
+
+        /// <summary>
+        /// .
+        /// </summary>
+        public ModelJsonConverter()
+            : this(true) { }
+
+        /// <summary>
+        /// .
+        /// </summary>
+        /// <param name="ignoreAdditionalProperties"></param>
+        public ModelJsonConverter(bool ignoreAdditionalProperties)
+        {
+            IgnoreAdditionalProperties = ignoreAdditionalProperties;
+        }
+
+        /// <summary>
+        /// .
+        /// </summary>
+        /// <param name="typeToConvert"></param>
+        /// <returns></returns>
+        public override bool CanConvert(Type typeToConvert)
+        {
+            return (typeToConvert.GetInterfaces().Any(i => i is IJsonSerializable));
+        }
+
+        /// <summary>
+        /// .
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="typeToConvert"></param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+        public override IJsonSerializable Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+        {
+            SerializableOptions serializableOptions = ConvertOptions(options);
+            var model = Activator.CreateInstance(typeToConvert, true) as IModelInternalSerializable;
+            if (model is null)
+                throw new NotSupportedException($"{typeToConvert.Name} does not have a parameterless constructor");
```

Comment:
won't `Activator.CreateInstance` already throw if a suitable constructor is not available? Seems like the error should be related to the not implementing `IModelInternalSerializable`